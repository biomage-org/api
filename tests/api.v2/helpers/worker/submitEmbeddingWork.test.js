const AWSMock = require('aws-sdk-mock');
const _ = require('lodash');

const createObjectHash = require('../../../../src/api.v2/helpers/worker/createObjectHash');
const submitEmbeddingWork = require('../../../../src/api.v2/helpers/worker/workSubmit/submitEmbeddingWork');
const validateAndSubmitWork = require('../../../../src/api.v2/events/validateAndSubmitWork');
const { mockS3GetObject } = require('../../../test-utils/mockAWSServices');


jest.mock('../../../../src/api.v2/helpers/worker/createObjectHash');
jest.mock('../../../../src/api.v2/helpers/pipeline/getPipelineStatus');
jest.mock('../../../../src/api.v2/helpers/worker/getWorkerStatus');
jest.mock('../../../../src/api.v2/events/validateAndSubmitWork');

const message = {
  experimentId: '6463cb35-3e08-4e94-a181-6d155a5ca570',
  taskName: 'configureEmbedding',
  input: {
    experimentId: '6463cb35-3e08-4e94-a181-6d155a5ca570',
    taskName: 'configureEmbedding',
    processName: 'qc',
    sampleUuid: '',
    uploadCountMatrix: false,
    authJWT: 'Bearer whatever',
    config: {
      embeddingSettings: {
        method: 'umap',
        methodSettings: {
          tsne: { perplexity: 30, learningRate: 200 },
          umap: { distanceMetric: 'cosine', minimumDistance: 0.3 },
        },
      },
      clusteringSettings: {
        method: 'louvain',
        methodSettings: { louvain: { resolution: 0.8 } },
      },
    },

  },
  output: {
    bucket: 'worker-results-development-000000000000',
    key: '0eabfedf-0efe-4abf-8725-7062c54ed5e1',
  },
  response: { error: false },
  pipelineVersion: 2,
  apiUrl: null,
};

describe('submitWorkEmbedding', () => {
  beforeEach(() => {
    AWSMock.restore();
    process.env.USE_CACHE = 'true';
  });

  it('submits the work and the ETag / params are correct', async () => {
    mockS3GetObject({ Body: '{}' });

    const ETag = await submitEmbeddingWork(message);

    expect(createObjectHash.mock.calls).toMatchSnapshot();
    expect(ETag).toMatchSnapshot();
    expect(validateAndSubmitWork).toBeCalledTimes(1);
  });

  it('submits the work and the ETag / params are correct', async () => {
    mockS3GetObject({ Body: '{}' });

    message.input.config.embeddingSettings.useSaved = true;

    const ETag = await submitEmbeddingWork(message);

    expect(ETag).toBeNull();
    expect(validateAndSubmitWork).toBeCalledTimes(1);
  });
});
