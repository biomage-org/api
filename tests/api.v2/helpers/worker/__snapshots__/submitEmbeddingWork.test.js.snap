// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submitWorkEmbedding submits the work and the ETag / params are correct 1`] = `Array []`;

exports[`submitWorkEmbedding submits the work and the ETag / params are correct 2`] = `"4f9c58d4db390500745d62b5f7ee181c"`;
